libraries/Microsoft.Bot.Builder.Dialogs/DialogEvents.cs

```diff
@@ -36,5 +36,10 @@ public class DialogEvents
         /// Event fired when there was an exception thrown in the system.
         /// </summary>
         public const string Error = "error";
+
+        /// <summary>
+        /// Event fired by LG library.
+        /// </summary>
+        public const string LGEvents = "lgEvents";
```

Comment:
this is a layer violation.  This constant should simply be in LAnguageGenerationAssembly in a class called LGDialogEvents

Reply:
But if the event is not registered in the Dialog package, the breaking points would not work.
ref: the original source code:
https://github.com/microsoft/botbuilder-dotnet/blob/cdaa26ca15db8e657a3b4c7bb77a76f5e18a0c91/libraries/Microsoft.Bot.Builder.Dialogs.Debugging/DialogDebugAdapter.cs#L158

The `_events` only contains the events in DialogEvents as default.
So, should I extend the DialogEvents class and add a "lgEvent" event, and pass it into the DialogDebugAdapter in "UseDebugger" middleware?

Reply:
Yes, there are 2 ways to resolve this.
1. You can pass in a custom array of _events when UseDebugger() is called.  
2.  we could give this a name which is generic.  What about Custom = "custom" ? Evaluation="Evaluation"?  
I think I prefer #2.  Is there any reason this needs to be called LGEvent?

---
In reply to: [479843093](https://github.com/microsoft/botbuilder-dotnet/pull/3791#discussion_r479843093) [](ancestors = 479843093)

Reply:
Prefer 2 too. I would change this part soon. Thanks.

Reply:
# fixed